iroh/src/database/mem.rs

```diff
@@ -1,121 +1,495 @@
-//! An in memory implementation of [BaoMap] and [BaoReadonlyDb], useful for
-//! testing and short lived nodes.
+//! A full in memory database for iroh-bytes
+//!
+//! Main entry point is [Database].
```

Comment:
why do we need the test db if we have this?

Reply:
Not sure about the name, see the discussion with @Frando. But I like this one because it is lock free and very simple. It is the smallest possible impl that can be used to share stuff via iroh-bytes.